kernel/cinstr.mli

```diff
@@ -0,0 +1,43 @@
+(************************************************************************)
+(*  v      *   The Coq Proof Assistant  /  The Coq Development Team     *)
+(* <O___,, *   INRIA - CNRS - LIX - LRI - PPS - Copyright 1999-2017     *)
+(*   \VV/  **************************************************************)
+(*    //   *      This file is distributed under the terms of the       *)
+(*         *       GNU Lesser General Public License Version 2.1        *)
+(************************************************************************)
+open Names
+open Constr
+open Cbytecodes
+
+(** This file defines the lambda code for the bytecode compiler. It has been
+extracted from Clambda.ml because of the retroknowledge architecture. *)
+
+type uint =
+  | UintVal of Uint31.t
+  | UintDigits of lambda array
+  | UintDecomp of lambda
+
+and lambda =
+  | Lrel          of Name.t * int
+  | Lvar          of Id.t
+  | Lprod         of lambda * lambda
+  | Llam          of Name.t array * lambda
+  | Llet          of Name.t * lambda * lambda
+  | Lapp          of lambda * lambda array
+  | Lconst        of pconstant
+  | Lprim         of pconstant * int (* arity *) * instruction * lambda array
```

Comment:
Do we want to split `instruction` into two types (say `instructions` and `builtins`) and use only one of these types here? I would hate to see what `Lprim (..., Kswitch ...)` would ended up being compiled to?

Reply:
Yes, this is what we do in the branch with primitive integers. But for this PR I reused the type returned by the retroknowledge, which is `instruction`. I hope it is acceptable as an intermediate state. If not, I can temporarily introduce a new type and a mapping to `instruction`, but the primitive operations are going to change, so I'd rather work directly on polishing the primitive integer branch.

Reply:
Sure. Is the branch available somewhere? So that I stop suggesting things already on that branch.

Reply:
https://github.com/maximedenes/coq/tree/primitive-integers should give you an idea, but I need to rebase it on this one.

Reply:
On that branch, `cinstr` is merged back `clambda.ml` (it was split because of retroknowledge dependency issues), see e.g. https://github.com/maximedenes/coq/blob/primitive-integers/kernel/clambda.ml#L22 for the new type of `Lprim`.